src/formatting.cpp

```diff
@@ -54,9 +55,15 @@ namespace cpptrace {
         void filter(std::function<bool(const stacktrace_frame&)> filter) {
             options.filter = filter;
         }
+        void transform(std::function<stacktrace_frame(stacktrace_frame&&)> transform) {
+            options.transform = std::move(transform);
+        }
 
-        std::string format(const stacktrace_frame& frame, detail::optional<bool> color_override = detail::nullopt) const {
+        std::string format(stacktrace_frame frame, detail::optional<bool> color_override = detail::nullopt) const {
             std::ostringstream oss;
+            if(options.transform) {
+                frame = options.transform(std::move(frame));
```

Comment:
Thoughts on doing this as `options.transform(frame);` and just taking a lvalue ref?

Reply:
Ultimately, the perf had no measurable difference between implementations, so I think this level of minutia is better to address only when a profiler says it's a problem.

For the transform fn itself, I opted for rvalue ref to both be explicit and ownership transfer. Personally, I like being consistent with the rvalue refs, such that transformation is a moving operation across the board, rather than output param in some places, move param in others.

Again, though, your project. :slightly_smiling_face:

Reply:
Thanks, sorry to dwell on minutia. I try to get the library interface stuff right since it's hard to change. I think I'm good with this API so I'll go ahead and merge it! Thanks!